src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1324,7 +1331,24 @@ public static bool HasIdentityConversion(TypeSymbol type1, TypeSymbol type2)
             Debug.Assert((object)type1 != null);
             Debug.Assert((object)type2 != null);
 
-            return type1.Equals(type2, TypeCompareKind.AllIgnoreOptions);
+            // PROTOTYPE(NullableReferenceTypes): Equals should return a value that indicates
+            // directly whether the types were equal or equal ignoring nullable modifiers.
+            if (!type1.Equals(type2, TypeCompareKind.AllIgnoreOptions))
+            {
+                return false;
+            }
+
+            if (!type1.Equals(type2, TypeCompareKind.AllIgnoreOptions | TypeCompareKind.CompareNullableModifiersForReferenceTypes | TypeCompareKind.UnknownNullableModifierMatchesAny))
+            {
+                if (useSiteDiagnostics == null)
+                {
+                    useSiteDiagnostics = new HashSet<DiagnosticInfo>();
```

Comment:
do we want to call this "useSiteDiagnostics" ?  
I think  useSiteDiagnostics is used to report diagnostics on broken/missing symbols. This seems to be just a regular diagnostics related to a conversion. (not attached to a symbol and reported at use site).

Reply:
I agree, this should probably not be `useSiteDiagnostics`. This was a simple approach to allow returning the nullability warnings from conversion methods using an existing mechanism rather than changing the signatures of many methods. I will add a `// PROTOTYPE` comment to revisit this approach later.